<launch>

<param name="use_sim_time" value="true"/>

<node pkg="ekf_test" type="EkfConverter" name="ekfConverter"/>

<node pkg="tf" type="static_transform_publisher" name="map_odom" args="0 0 0 0 0 0 map odom 20"/>

<node pkg="tf" type="static_transform_publisher" name="map_ekfMap" args="0 0 0 0 0 0 ekfMap map 20"/>

<node pkg="hector_trajectory_server" type="hector_trajectory_server" name="odomTraj">
	<param name="target_frame_name" value="map" />
	<param name="source_frame_name" value="baseFootprint" />
	<param name="trajectory_update_rate" value="10" />
	<param name="trajectory_publish_rate" value="10" />
	<remap from="trajectory" to="trjOdom" />
</node>
	
<node pkg="hector_trajectory_server" type="hector_trajectory_server" name="ekfTraj">
	<param name="target_frame_name" value="map" />
	<param name="source_frame_name" value="ekfBaseFootprint" />
	<param name="trajectory_update_rate" value="10" />
	rvizEkfConverter<param name="trajectory_publish_rate" value="10" />
	<remap from="trajectory" to="trjEkf" />
</node>

<include file="$(find ekf_test)/launch/ekfRelative.launch"/>

<include file="$(find ekf_test)/launch/ekfAbsolute.launch"/>

<node pkg="rosbag" type="play" name="rosbag_play" output="screen" args="--clock -r 2 -s 2 $(find ekf_test)/bag/boschKupferspuleCeilCam.bag"/>

<node pkg="rviz" type="rviz" name="rviz" args="-d $(find ekf_test)/cfg/EkfConverter.rviz" />

</launch>